File: family/src/main/java/com/jingxi/smartlife/pad/family/mvp/data/EquipmentBean.java
package com.jingxi.smartlife.pad.family.mvp.data;

import androidx.annotation.IdRes;

import java.util.ArrayList;
import java.util.List;

/**
 * author ：shenbingkai@163.com
 * date ：2019-01-23
 * description ：
 */
public class EquipmentBean {

    public EquipmentType type;
    public String title;
    @IdRes
    public int icon;
    public List<Menu> menus;

    public EquipmentBean(EquipmentType type) {
        this.type = type;
    }

    public EquipmentBean(EquipmentType type, String title, int icon) {
        this.type = type;
        this.title = title;
        this.icon = icon;
    }

    public EquipmentBean addMenu(Menu menu) {
        if (null == menus)
            menus = new ArrayList<>();

        menus.add(menu);
        return this;
    }

    public static class Menu {
        public String title;
        @IdRes
        public int icon;

        public Menu(String title, int icon) {
            this.title = title;
            this.icon = icon;
        }
    }
}

File: family/src/main/java/com/jingxi/smartlife/pad/family/mvp/data/EquipmentBean.kt
package com.jingxi.smartlife.pad.family.mvp.data

import androidx.annotation.IdRes
import java.util.*

/**
 * author ：shenbingkai@163.com
 * date ：2019-01-23
 * description ：
 */
class EquipmentBean {

    var type: EquipmentType
    var title: String?=null
    @IdRes
    var icon: Int = 0
    var menus: MutableList<Menu>? = null

    constructor(type: EquipmentType, title: String, icon: Int) {
        this.type = type
        this.title = title
        this.icon = icon
    }

    fun addMenu(menu: Menu): EquipmentBean {
        if (null == menus)
            menus = ArrayList()

        menus!!.add(menu)
        return this
    }

    class Menu(var title: String, @field:IdRes
    var icon: Int)
}
